sdks/java/core/src/main/java/org/apache/beam/sdk/coders/VoidCoder.java

```diff
@@ -38,7 +39,10 @@ public static VoidCoder of() {
   private VoidCoder() {}
 
   @Override
-  public void encode(Void value, OutputStream outStream) {
+  public void encode(Void value, OutputStream outStream) throws IOException {
+    if (value != null) {
+      throw (new IOException("Attempting to encode non-null value with VoidCoder."));
```

Comment:
Drop extraneous brackets:
`throw (new IOException("Attempting to encode non-null value with VoidCoder."));` -> `throw new IOException("Attempting to encode non-null value with VoidCoder.");`